src/ReplicaManager.ts

```diff
@@ -1,27 +1,26 @@
 import { PrismaClient } from '@prisma/client/extension'
-
-type PrismaConstructorOptions = {
-  datasourceUrl?: string
-}
+import { PrismaClientOptions } from '@prisma/client/runtime/library'
 
 export type ConfigureReplicaCallback = (client: PrismaClient) => PrismaClient
 interface PrismaClientConstructor {
-  new (options?: PrismaConstructorOptions): PrismaClient
+  new (options?: PrismaClientOptions): PrismaClient
```

Comment:
That type might not actually correspond to what's getting generated - this one is internal and includes every possible option that can exists, the one that will be generated might be missing some of the options depending on priovider/preview features used.